modules/openapi-generator/src/main/resources/csharp-netcore/libraries/generichost/api.mustache

```diff
@@ -0,0 +1,324 @@
+        
+        /// <summary>
+        /// {{summary}}
+        /// </summary>
+        /// <remarks>
+        /// {{notes}}
+        /// </remarks>
+        {{#allParams}}
+        /// <param name="{{paramName}}">{{description}}{{^required}} (optional{{#defaultValue}}, default to {{.}}{{/defaultValue}}){{/required}}</param>
+        {{/allParams}}
+        /// <param name="cancellationToken">Cancellation Token to cancel the request.</param>
+        /// <returns>Task of ApiResponse&lt;{{#returnType}}{{returnType}}{{/returnType}}{{^returnType}}object{{/returnType}}{{#nullableReferenceTypes}}?{{/nullableReferenceTypes}}&gt;</returns>
+        Task<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}object{{/returnType}}{{#nullableReferenceTypes}}?{{/nullableReferenceTypes}}> {{operationId}}OrDefaultAsync({{#allParams}}{{#required}}{{{dataType}}} {{paramName}}{{^-last}}, {{/-last}}{{/required}}{{^required}}{{{dataType}}} {{paramName}} = null{{^-last}}, {{/-last}}{{/required}}{{/allParams}}{{#allParams.0}}, {{/allParams.0}}System.Threading.CancellationToken{{#nullableReferenceTypes}}?{{/nullableReferenceTypes}} cancellationToken = null);
+
+        {{/operation}}
+    }
+
+    /// <summary>
+    /// Represents a collection of functions to interact with the API endpoints
+    /// </summary>
+    {{>visibility}} partial class {{classname}} : {{interfacePrefix}}{{classname}}
+    {
+        /// <summary>
+        /// An event to track the health of the server. 
+        /// If you store these event args, be sure to purge old event args to prevent a memory leak.
+        /// </summary>
+        public event ClientUtils.EventHandler<ApiResponseEventArgs>? ApiResponded;
+
+        private readonly ILogger<{{classname}}> _logger;
+        private readonly HttpClient _httpClient;{{#hasApiKeyMethods}}
+        private readonly TokenProvider<ApiKeyToken> _apiKeyProvider;{{/hasApiKeyMethods}}{{#hasHttpBearerMethods}}
+        private readonly TokenProvider<BearerToken> _bearerTokenProvider;{{/hasHttpBearerMethods}}{{#hasHttpBasicMethods}}
+        private readonly TokenProvider<BasicToken> _basicTokenProvider;{{/hasHttpBasicMethods}}{{#hasHttpSignatureMethods}}
+        private readonly TokenProvider<HttpSignatureToken> _httpSignatureTokenProvider;{{/hasHttpSignatureMethods}}{{#hasOAuthMethods}}
+        private readonly TokenProvider<OAuthToken> _oauthTokenProvider;{{/hasOAuthMethods}}
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="{{classname}}"/> class.
+        /// </summary>
+        /// <returns></returns>
+        public {{classname}}(ILogger<{{classname}}> logger, HttpClient httpClient, {{#hasApiKeyMethods}}
```

Comment:
Dangling comma in case all api key methods are false

Reply:
Fixed